nupic/research/frameworks/vernon/mixins/track_representation_sparsity.py

```diff
@@ -0,0 +1,129 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+from contextlib import nullcontext
+from pprint import pformat
+
+from nupic.research.frameworks.pytorch.hooks import ModelHookManager, TrackSparsityHook
+
+
+class TrackRepresentationSparsity:
+    """
+    This mixin tracks and reports the average sparsities observed in the model's
+    representations. Either input or output sparsities can be tracked for a specified,
+    possibly overlapping, subset of modules. Tracked statistics are returned from
+    `run_epoch` and are reset before each subsequent epoch. The default is to track none
+    of the modules; only those specified can be tracked.
+
+    .. _filter_modules: nupic.research.frameworks.pytorch.model_utils.filter_modules
+    """
+
+    def setup_experiment(self, config):
+        """
+        Register forward hooks to track the input and output sparsities.
+
+        The subsets of modules to track are defined via `include_*` params. See
+        `filter_modules`_ for further details.
+
+        :param config:
+            - track_input_sparsity_args:
+                - include_modules a list of module types to track
```

Comment:
Adding a : to separate the argument name and its description will make it more clear